### ISSUE TITLE
conda build 2.0.0 - filelock fail on missing pkgs directory

### ISSUE THREAD

**Issue by bjlittle** [conda/conda-build#1296]
Working with [conda build](https://github.com/conda/conda-build/tree/242e5accccea9196c58f10cb4ea7de21286b6b7b/conda_build):

```
version_version = '2.0.0'
version_full = '242e5accccea9196c58f10cb4ea7de21286b6b7b'
```

And getting the following traceback:
```
Traceback (most recent call last):
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/bin/conda-build-all", line 11, in <module>
    load_entry_point('conda-build-all==1.0.0rc1', 'console_scripts', 'conda-build-all')()
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build_all/cli.py", line 90, in main
    b.main()
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build_all/builder.py", line 263, in main
    built_dist_location = self.build(meta, build_config)
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build_all/builder.py", line 200, in build
    conda_build.api.build(meta.meta, config=config)
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build/api.py", line 81, in build
    need_source_download=need_source_download, config=config)
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build/build.py", line 942, in build_tree
    config=config)
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build/build.py", line 546, in build
    create_env(config.build_prefix, specs, config=config)
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/conda_build/build.py", line 431, in create_env
    lock.acquire(timeout=config.timeout)
  File "/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/lib/python3.5/site-packages/filelock.py", line 243, in acquire
    raise Timeout(self._lock_file)
filelock.Timeout: The file lock '/var/tmp/itwl/conda/tmp_envs/bd0dda3688f45d0a2adf/pkgs/.conda_lock' could not be acquired.
```
The fail appears to be related to the fact that within `create_env` it is attempting to acquire a lock on a file within  the [dirname = os.path.join(cc.root_dir, 'pkgs')](https://github.com/conda/conda-build/blob/242e5accccea9196c58f10cb4ea7de21286b6b7b/conda_build/build.py#L427) directory and that `pkgs` root sub-directory does **not** exist.

Would it be valid for `create_env` to create the `pkgs` sub-directory if it doesn't exist, before attempting to acquire the filelock? Or is this issue the symptom of some other underlying problem?

Many thanks!

**Comment by msarahan** [comment]
Thanks for reporting this.  I have added a fix in #1299, and this will go out in the next bugfix release later today or tomorrow.

**Comment by bjlittle** [comment]
Awesome, thanks @msarahan :smile:
